### DISCUSSION TITLE
New streams.<key>.where to avoid using streams.<key>.sql

### DISCUSSION THREAD

**Discussion by CharlesRngrd** [slingdata-io/sling-cli#381] [Ideas]
Hello,

I'm always using a basic `select/from/where` SQL request in my stream.

Here is a typical stream:

```yaml
my_stream:
  description: Journal des stocks
  sql: |
    select
      itmref_0,
      iptdat_0,
      qtypcu_0,
      priord_0,
      vcrtyp_0,
      amtord_0,
      stofcy_0
    from stojou
    where extract(year from iptdat_0) >= 2023
```

Here is what I would like to write i:

```yaml
  my_stream:
    description: Journal des stocks
    select:
      itmref_0
      iptdat_0
      qtypcu_0
      priord_0
      vcrtyp_0
      amtord_0
      stofcy_0
    object: stojou
    where: extract(year from iptdat_0) >= 2023
```

This will allow me to tell my team members to not use custom SQL (because I would like to use Sling for EL and DBT for T).

Both `select `and `object `are implemented but not `where`.

Could you implement it please ?

Have a nice day

**Comment by flarco** [comment]
Hi @CharlesRngrd, very late, but the `where` input is added, can you try with the [dev build - 1.3.6.dev](https://github.com/slingdata-io/sling-cli?tab=readme-ov-file#automated-dev-builds) and report any issues?

**Comment by CharlesRngrd** [comment]
Great, thank you @flarco !

I'm going to try it out.
If anything goes wrong I will come back to you.
